core/dbt/contracts/graph/parsed.py

```diff
@@ -810,14 +835,38 @@ class ParsedMetric(UnparsedBaseNode, HasUniqueID, HasFqn):
     filters: List[MetricFilter]
     time_grains: List[str]
     dimensions: List[str]
+    model: Optional[str] = None
     resource_type: NodeType = NodeType.Metric
     meta: Dict[str, Any] = field(default_factory=dict)
     tags: List[str] = field(default_factory=list)
     sources: List[List[str]] = field(default_factory=list)
     depends_on: DependsOn = field(default_factory=DependsOn)
     refs: List[List[str]] = field(default_factory=list)
+    metrics: List[List[str]] = field(default_factory=list)
+    # ratio_terms: Optional[ParsedMetricRatioTerms] = None
+    ratio_terms: Optional[MetricRatioTerms] = None
+    is_derived: bool = False
     created_at: float = field(default_factory=lambda: time.time())
 
+    def resolve_metric_references(self, context):
+        # TODO: Obviously don't do this...
```

Comment:
Yes, exposures don't refer to themselves, so exposures refs, sources, etc get resolved in the various parser.manifest 'process_XXX' methods.  I wouldn't call what the metrics need a "patch" step, because when we did that we saved a bunch of patch dictionaries (with config, etc) until the end of parsing and applied them then. So far metrics just need a resolution phase, as you've implemented.